subsystems/drivebase/SwerveDrive.java

```diff
@@ -59,6 +61,10 @@ public abstract class SwerveDrive extends DriveBase {
     private final Mk4ModuleConfiguration kModuleConfigBackRight = new Mk4ModuleConfiguration();
     private final double kPitchOffset;
     private final double kRollOffset;
+    private TalonFXConfiguration driveMotorConfig = new TalonFXConfiguration();
+    private CurrentLimitsConfigs currentLimitConfigs = driveMotorConfig.CurrentLimits;
+    private final Queue<Double> driveCurrentBuffer;
```

Comment:
could we not just use a fixed-size array of doubles and keep an index to keep track of where we are in the array? I'm still pushing to make this not create garbage if it doesn't have to

Reply:
Wouldn't things get weird with that because it would be hard to remove the oldest entry and add a new one since the entire array would have to shift? And that's not really feasible with plain arrays? Not sure how to go about it but I might be thinking about it wrong.

Reply:
No, you wouldn't shift. This kind of data structure is called a ring buffer. See https://www.baeldung.com/cs/circular-buffer for example. You'd keep track of the next index in the array to add the next entry. Then when you add an entry, you do something like:

if the buffer is full then we're overwriting the value at buffer[index] so grab its current value so we can recalc the rolling average

then overwrite it:
buffer[index] = newValue;

then update the index:
index = (index + 1) % BUFFER_SIZE;

and check if the buffer is now full (if we've rolled over to 0, it's full):
if (index == 0) {
  bufferFull = true;
}

or something like that

Does that make sense? Happy to talk about it. And if you think this is too much, that's fine, too. We're maybe squeezing performance that doesn't need to be squeezed. But I don't think it's that complicated and I do worry a little bit about garbage collection - we're doing more and more each year and the roborio doesn't have that much memory.